rules/Privatization/foo_bar.php

```diff
@@ -0,0 +1,3 @@
+<?php
```

Comment:
it needs 2 files:

1. `foo*bar.php`
2. `foo_bar.php`

Reply:
on windows I am not allowed to create a file which contains a `*`.

Reply:
then, need to be tested with run:

```
bin/rector process rules/Privatization/foo*bar.php --dry-run
```

before middle name check: 1 files processed (expected)
after middle name check: 2 files processed (not expected)

Reply:
I don't agree. with a skip pattern of `__DIR__ . '/rules/Privatization/foo*bar.php',` I would expect the `*` to resolve to any character, like it does in all other patterns you can find in the `rector.php`.

why should it sometimes only resolve to the concrete file-name?

Reply:
`'*/Fixture/*',` does not resolve to a directory named `*` but to all directories which can be found below a directory named `Fixture`

Reply:
On command line, the following command:

```bash
bin/rector process rules/Privatization/foo*bar.php --dry-run
```

got 2 files actually, they are:

1. `foo*bar.php`
2. `foo_bar.php`
 
but skip the `foo*bar.php` only which the file exists, if you want to skip the `foo_bar.php` as well, you need to exclude `__DIR__` in the config:

```php
        'rules/Privatization/foo*bar.php',
```

so the `fnmatch()` will working.

Reply:
I don't agree.

Reply:
The `fnmatch()` never work on real path, only on masking path, otherwise, it will have different behaviour between on cli path vs from config paths

Reply:
see https://3v4l.org/Gpod7 for the reasoning